store/sqlstore/plugin_store.go

```diff
@@ -71,6 +71,43 @@ func (ps SqlPluginStore) SaveOrUpdate(kv *model.PluginKeyValue) store.StoreChann
 	})
 }
 
+func (ps SqlPluginStore) CompareAndUpdate(kv *model.PluginKeyValue, oldValue []byte) store.StoreChannel {
+	return store.Do(func(result *store.StoreResult) {
```

Comment:
We're in the middle of a newly launched campaign to simplify our store API to just blocking calls -- since most of the time we use them as such, and can easily parallelize in the client code.

Can you adopt that pattern pre-emptively here? See https://github.com/mattermost/mattermost-server/pull/10633 to illustrate.